cli/tsc/99_main_compiler.js

```diff
@@ -1328,18 +1327,73 @@ delete Object.prototype.__proto__;
     }
   }
 
-  let hasBootstrapped = false;
+  /**
+   * @typedef {object} Request
+   * @property {Record<string, any>} config
+   * @property {boolean} debug
+   * @property {string[]} rootNames
+   */
 
-  function bootstrapCompilerRuntime({ debugFlag }) {
-    if (hasBootstrapped) {
-      throw new Error("Worker runtime already bootstrapped");
+  /** The API that is called by Rust when executing a request.
+   * @param {Request} request 
+   */
+  function exec({ config, debug: debugFlag, rootNames }) {
+    setLogDebug(debugFlag, "TS");
+    performanceStart();
+    debug(">>> exec start", { rootNames });
+    debug(config);
+
+    const { options, errors: configFileParsingDiagnostics } = ts
+      .convertCompilerOptionsFromJson(config, "", "tsconfig.json");
```

Comment:
No problems with `allowNonTsExtensions` setting? I had to explicitly override it in each of legacy handlers like so: `options.allowNonTsExtensions = true;`

Reply:
No problems, because the compiler doesn't resolve any extensions, it never sees an extension... it is fed 100% of the extensions via the `resolveModules()`.  I was wondering why you were struggling with this when trying to adapt stuff from my prototype, but I believe that is the reason.

Reply:
Great! That felt wrong anyway